### ISSUE TITLE
Questions: frontmatter variables, migration, and inheritance

### ISSUE THREAD

**Issue by fuzzah** [silverbulletmd/silverbullet#666]
New user here, may be misunderstanding something ;(

### Can we include frontmatter variables in the same page?
Something like this:
````md
---
tags: book
title: My title
author: Some author
---
# My review for the book {{title} by {{author}}
|^|

# Other books by this author
```query
book where author =~ /{{author}}/ render [[templates/book]]
```
````
Of course the point is to update them only in the frontmatter without having to change them everywhere on the page.

### Can we somehow migrate pages to use newer frontmatter after template update?
If I had variables like `title` and `author` in the book template, and then decided to add something like `rating` or rename `author` to `authors`, I would have to manually update 100+ books I have. There's ought to be at least some helper mechanisms to migrate the old pages. We could at least use some warning like "newer template has these frontmatter changes: added field X, removed field Y, renamed field A to B, ..."

### Is there some sort of frontmatter inheritance?
I'd like to define generic books, but then separate them to technical books, fiction books, etc. Inherited templates would then include some additional fields based on their kind, while having their base fields reused (and synchronized) from the parent template.

**Comment by zefhemel** [comment]
> Can we include frontmatter variables in the same page?

Yes, but in a slightly ugly way. You can use a template for this:

    ```template
    template: |
        # My review for the book {{title}} by {{author}}
    ```

> Can we somehow migrate pages to use newer frontmatter after template update?

There's no real tools for this yet, but feel free to a GH issue to request this. However, since this is all markdown files on a disk, you can use any other tools to achieve this. If you're comfortable with VS Code, you may want to do a folder-wide search and replace for instance.

> Is there some sort of frontmatter inheritance?

Do you have a concrete example? You're looking to inherit actual data?
